File: Hotels/Models/Reservation.cs
﻿using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Hotels.Models
{
    public class Reservation
    {
        public int Id { get; set; }

        [Range(0, 100)]
        public int Discount { get; set; }

        [Required]
        [Display(Name = "Start date")]
        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:dd-MM-YYYY}")]
        public DateTime StartDate { get; set; }

        [Required]
        [Display(Name = "End date")]
        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:dd-MM-YYYY}")]
        public DateTime EndDate { get; set; }

        [ForeignKey("Guest")]
        [Display(Name = "Guest ")]
        public int GuestId { get; set; }
        public virtual Guest Guest { get; set; }

        [ForeignKey("Room")]
        [Display(Name = "Room")]
        public int RoomId { get; set; }
        public virtual Room Room { get; set; }

        public virtual ReservationStatus ReservationStatus { get; set; }
    }
}

File: Hotels/Scripts/Reservation.js
function Reservation(object) {
    if (object) {
        this.id = object.Id;
        this.startDate = object.StartDate;
        this.endDate = object.EndDate;
        this.roomId = object.RoomId;
        this.guestId = object.GuestId;
        this.discount = object.discount
    } else {
        this.id = null;
        this.startDate = null;
        this.endDate = null;
        this.roomId = null;
        this.guestId = null;
        this.discount = null;
    }
}

Reservation.prototype.toString = function () {
    return this.id + ": " + this.startDate + " - " + this.endDate;
}